src/wallet/wallet.cpp

```diff
@@ -3104,6 +3104,56 @@ std::shared_ptr<CWallet> CWallet::Create(WalletContext& context, const std::stri
     return walletInstance;
 }
 
+std::shared_ptr<CWallet> CWallet::LoadExisting(WalletContext& context, const std::string& name, std::unique_ptr<WalletDatabase> database, uint64_t wallet_creation_flags, bilingual_str& error, std::vector<bilingual_str>& warnings)
+{
+    interfaces::Chain* chain = context.chain;
+    const std::string& walletFile = database->Filename();
+
+    const auto start{SteadyClock::now()};
+    std::shared_ptr<CWallet> walletInstance(new CWallet(chain, name, std::move(database)), FlushAndDeleteWallet);
+
+    if (!LoadWalletArgs(walletInstance, context, error, warnings)) {
+        return nullptr;
+    }
+
+    // Load wallet
+    bool rescan_required = false;
+    auto nLoadWalletRet = walletInstance->PopulateWalletFromDB(error, warnings);
+    if (nLoadWalletRet == DBErrors::NEED_RESCAN) {
+        rescan_required = true;
+    } else if (nLoadWalletRet != DBErrors::LOAD_OK && nLoadWalletRet != DBErrors::NONCRITICAL_ERROR) {
+        return nullptr;
+    }
```

Comment:
In 4c6890cdae77ca0d230227530b000f4b3c6fd77a "wallet: Create separate function for wallet load"

I think this can be simplified:

```suggestion
    auto nLoadWalletRet = walletInstance->PopulateWalletFromDB(error, warnings);
    bool rescan_required = nLoadWalletRet == DBErrors::NEED_RESCAN;
    if (nLoadWalletRet != DBErrors::LOAD_OK && nLoadWalletRet != DBErrors::NONCRITICAL_ERROR && !rescan_required) {
        return nullptr;
    }
```